docs/CODE_GUIDELINES.md

```diff
@@ -3,193 +3,212 @@
-```cpp
-std::vector<std::string> test; test.push_back("foobar");
-```
-Always use a new line for a new statement.
+## 3. Formatting
+
+### 3.1. Braces
+Curly braces always go on a new line.
 ```cpp
-std::vector<std::string> test;
-test.push_back("foobar");
+for (int i = 0; i < t; i++)
+{
+  [...]
+}
+
+if (true)
+{
+  [...]
+}
+
+class Dummy
+{
+  [...]
+};
 ```
-It is much easier to debug if one can pinpoint a precise line number.
 
-**[back to top](#table-of-contents)**
+### 3.2. Indentation
+Use spaces as tab policy with an indentation size of 2.
+Opening curly braces increase the level of indentation. Closing curly braces decrease the level of indentation.
+
+**Exception:** Do not indent namespaces to simplify nesting them and wrapping *.cpp* files in a namespace.
 
-## 5. Namespaces
-Indentation is not required to simplify nested namespaces and wrapping *.cpp* files in a namespace.
 ```cpp
 namespace KODI
 {
 namespace UTILS
 {
+
 class ILogger
 {
-  void Log(...) = 0;
+public:
+  virtual void Log(...) = 0;
+  virtual ~ILogger() {}
 }
+
 }
 }
 ```
 
-**[back to top](#table-of-contents)**
+### 3.3. Control statements
+Insert a new line before every:
+* else in an if statement
+* catch in a try statement
+* while in a do statement
 
-## 6. Headers
-Included header files have to be sorted alphabetically to prevent duplicates and allow better overview, with an empty line clearly separating sections.
-
-Header order has to be:
-* Own header file
-* Other Kodi includes
-* C and C++ system files
-* Other libraries' header files
+#### 3.3.1. if else
+Put the consequent on a new line if not in curly braces anyway. Keep `else if` statements on one line.
 
 ```cpp
-#include "PVRManager.h"
-
-#include "addons/AddonInstaller.h"
-#include "dialogs/GUIDialogExtendedProgressBar.h"
-#include "messaging/helpers/DialogHelper.h"
-#include "messaging/ApplicationMessenger.h"
-#include "messaging/ThreadMessage.h"
-#include "music/tags/MusicInfoTag.h"
-#include "music/MusicDatabase.h"
-#include "network/Network.h"
-#include "pvr/addons/PVRClients.h"
-#include "pvr/channels/PVRChannel.h"
-#include "settings/Settings.h"
-#include "threads/SingleLock.h"
-#include "utils/JobManager.h"
-#include "utils/log.h"
-#include "utils/Variant.h"
-#include "video/VideoDatabase.h"
-#include "Application.h"
-#include "ServiceBroker.h"
-
-#include <cassert>
-#include <utility>
-
-#include <libavutil/pixfmt.h>
+if (true)
+  return;
+if (true)
+{
+  [...]
+} 
+else if (false)
+{
+  return;
+} 
+else
+  return;
 ```
 
-Place directories before files. If the headers aren't sorted, either do your best to match the existing order, or precede your commit with an alphabetization commit.
-
-If possible, avoid including headers in another header. Instead, you can forward-declare the class and use a `std::unique_ptr`:
-
+#### 3.3.2. switch case
 ```cpp
-class CFileItem;
-
-class Example
+switch (cmd)
 {
-  ...
-  std::unique_ptr<CFileItem> m_fileItem;
+  case x:
```

Comment:
In many parts of the code, `case` isn't indented to reduce the number of nested indentations